[CI] RandomSamplerAggregatorTests testAggregationSamplingNestedAggsScaled failing

I was able to reproduce this on OSx via 
```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.sampler.random.RandomSamplerAggregatorTests.testAggregationSamplingNestedAggsScaled" -Dtests.seed=B7AFBEB0142797C6 -Dtests.locale=nl-BE -Dtests.timezone=Atlantic/Azores
```

**Build scan:**
https://gradle-enterprise.elastic.co/s/mkzxxnxywqajw/tests/:server:test/org.elasticsearch.search.aggregations.bucket.sampler.random.RandomSamplerAggregatorTests/testAggregationSamplingNestedAggsScaled

**Reproduction line:**
`gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.sampler.random.RandomSamplerAggregatorTests.testAggregationSamplingNestedAggsScaled" -Dtests.seed=B7AFBEB0142797C6 -Dtests.locale=nl-BE -Dtests.timezone=Atlantic/Azores -Druntime.java=18`

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.search.aggregations.bucket.sampler.random.RandomSamplerAggregatorTests&tests.test=testAggregationSamplingNestedAggsScaled

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a value greater than <0L>
     but: <0L> was equal to <0L>

  at __randomizedtesting.SeedInfo.seed([B7AFBEB0142797C6:859F02F8A1FFCAC2]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.search.aggregations.bucket.sampler.random.RandomSamplerAggregatorTests.lambda$testAggregationSamplingNestedAggsScaled$1(RandomSamplerAggregatorTests.java:100)
  at org.elasticsearch.search.aggregations.AggregatorTestCase.testCase(AggregatorTestCase.java:676)
  at org.elasticsearch.search.aggregations.bucket.sampler.random.RandomSamplerAggregatorTests.testAggregationSamplingNestedAggsScaled(RandomSamplerAggregatorTests.java:82)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:577)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/server/src/test/java/org/elasticsearch/search/aggregations/bucket/sampler/random/RandomSamplerAggregatorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/bucket/sampler/random/RandomSamplerAggregatorTests.java
@@ -94,10 +94,18 @@ public class RandomSamplerAggregatorTests extends AggregatorTestCase {
                 long outerFilterDocCount = agg.getDocCount();
                 Filter innerAgg = agg.getAggregations().get("filter_inner");
                 long innerFilterDocCount = innerAgg.getDocCount();
-                // subaggs should be scaled along with upper level aggs
-                assertThat(outerFilterDocCount, equalTo(innerFilterDocCount));
-                // sampled doc count is NOT scaled, and thus should be lower
-                assertThat(outerFilterDocCount, greaterThan(sampledDocCount));
+                if (sampledDocCount == 0) {
+                    // in case 0 docs get sampled, which can rarely happen
+                    // in case the test index has many segments.
+                    assertThat(sampledDocCount, equalTo(0L));
+                    assertThat(innerFilterDocCount, equalTo(0L));
+                    assertThat(outerFilterDocCount, equalTo(0L));
+                } else {
+                    // subaggs should be scaled along with upper level aggs
+                    assertThat(outerFilterDocCount, equalTo(innerFilterDocCount));
+                    // sampled doc count is NOT scaled, and thus should be lower
+                    assertThat(outerFilterDocCount, greaterThan(sampledDocCount));
+                }
             },
             longField(NUMERIC_FIELD_NAME),
             keywordField(KEYWORD_FIELD_NAME)

